src/sonic-bgpcfgd/bgpcfgd/managers_bbr.py

```diff
@@ -0,0 +1,120 @@
+                self.constants['bgp']['bbr']['enabled']:
+            self.bbr_enabled_pgs = self.__read_pgs()
+            if self.bbr_enabled_pgs:
+                self.enabled = True
+                self.directory.put(self.db_name, self.table_name, 'status', "enabled")
+                log_info("BBRMgr::Initialized and enabled")
+            else:
+                log_info("BBRMgr::Disabled: no BBR enabled peers")
+        else:
+            log_info("BBRMgr::Disabled: not enabled in the constants")
+
+    def __read_pgs(self):
+        """
+        Read peer-group bbr settings from constants file
+        :return: return bbr information from constant peer-group settings
+        """
+        if 'peers' not in self.constants['bgp']:
+            log_info("BBRMgr::no 'peers' was found in constants")
+            return {}
+        res = {}
+        for peer_name, value in self.constants['bgp']['peers'].items():
+            if 'bbr' not in value:
+                continue
+            for pg_name, pg_afs in value['bbr'].items():
+                res[pg_name] = pg_afs
+        return res
+
+    def __set_validation(self, key, data):
+        """ Validate set-command arguments
+        :param key: key of 'set' command
+        :param data: data of 'set' command
+        :return: True is the parameters are valid, False otherwise
+        """
+        if key != 'all':
+            log_err("Invalid key '%s' for table '%s'. Only key value 'all' is supported" % (key, self.table_name))
+            return False
+        if 'status' not in data:
+            log_err("Invalid value '%s' for table '%s', key '%s'. Key 'status' in data is expected" % (data, self.table_name, key))
+            return False
+        if data['status'] != "enabled" and data['status'] != "disabled":
+            log_err("Invalid value '%s' for table '%s', key '%s'. Only 'enabled' and 'disabled' are supported" % (data, self.table_name, key))
+            return False
+        return True
+
+    def __set_prepare_config(self, status):
+        """
+        Generate FFR configuration to apply changes
+        :param status: either "enabled" or "disabled"
+        :return: list of commands prepared for FRR
+        """
+        bgp_asn = self.directory.get_slot("CONFIG_DB", swsscommon.CFG_DEVICE_METADATA_TABLE_NAME)["localhost"]["bgp_asn"]
+        cmds = ["router bgp %s" % bgp_asn]
+        prefix_of_commands = "" if status == "enabled" else "no "
+        for af in ["ipv4", "ipv6"]:
+            cmds.append(" address-family %s" % af)
+            for pg_name in sorted(self.bbr_enabled_pgs.keys()):
+                if af in self.bbr_enabled_pgs[pg_name]:
+                    cmds.append("  %sneighbor %s allowas-in 1" % (prefix_of_commands, pg_name))
+        return cmds
+
+    def __restart_peers(self):
+        """ Restart peer-groups which support BBR """
+        for peer_group in sorted(self.bbr_enabled_pgs.keys()):
```

Comment:
why sort here? any reason?

Reply:
Otherwise my test doesn't know what to expect (or I can use conversion to sets) and compare sets. I'm going to remove it after the batching change